Extend metadata in details panel for Weave Net nodes

2opremio opened the pull request:
Fixes #1938 

Here's how it looks:

<img width="2108" alt="screen shot 2016-11-03 at 16 04 29" src="https://cloud.githubusercontent.com/assets/2362916/19973937/0c80fcb4-a1e7-11e6-9a94-bb2b016260f5.png">




There's still no connections table, that will come later. I don't see it critical for the kubecon deadline, but let me know if I am wrong.

rade commented:
> There's still no connection table...  let me know if I am wrong.

You are wrong.

2opremio commented:
> You are wrong.

OK, I will work on that next, but let's merge this first

rade commented:
I find the "#<num>" notation a bit odd. I don't think we are using that elsewhere.

jml commented on probe/overlay/weave.go:
```diff
@@ -1,61 +1,152 @@
+	WeaveProxyAddress      = "weave_proxy_address"
+	WeavePluginTableID     = "weave_plugin_table"
+	WeavePluginStatus      = "weave_plugin_status"
+	WeavePluginDriver      = "weave_plugin_driver"
```
lolgo

jml commented on probe/overlay/weave.go:
```diff
@@ -1,61 +1,152 @@
+	weaveMetadata = report.MetadataTemplates{
+		WeaveVersion:        {ID: WeaveVersion, Label: "Version", From: report.FromLatest, Priority: 1},
+		WeaveProtocol:       {ID: WeaveProtocol, Label: "Protocol", From: report.FromLatest, Priority: 2},
+		WeavePeerName:       {ID: WeavePeerName, Label: "Peer Name", From: report.FromLatest, Truncate: 17, Priority: 3},
```
Why is this truncated? Why 17? (Please add a comment)

2opremio replied:
No real reason, I will remove the truncation.

jml commented on probe/overlay/weave.go:
```diff
@@ -106,25 +207,98 @@ func (w *Weave) status() (bool, error) {
+
+	switch err.(type) {
+	case *docker_client.Error:
+		// This is really ugly, but there's no way around it :(
```
Maybe say why there's no way around it?

2opremio replied:
Sure

jml commented on probe/overlay/weave.go:
```diff
@@ -162,43 +336,120 @@ func (w *Weave) Report() (report.Report, error) {
 	return r, nil
 }
+
+func (w *Weave) getPeerNode(peer weave.Peer) report.Node {
```
Probably worth adding a doc comment for this.

2opremio replied:
Sure

jml commented on prog/probe.go:
```diff
@@ -40,7 +40,10 @@ const (
-var pluginAPIVersion = "1"
+var (
+	pluginAPIVersion = "1"
+	dockerEndpoint   = "unix:///var/run/docker.sock"
```
Shouldn't this get `DOCKER_HOST` or something, to handle docker-machine / minikube cases?

2opremio replied:
Maybe, not part in this PR though. I was nice enough to refactor it a bit :)

I have created https://github.com/weaveworks/scope/issues/1975

2opremio replied:
> to handle docker-machine / minikube cases?

However, this won't be the usecase since scope also needs to live in the machine being monitored (it needs `/proc` and conntrack for instance)

jml commented on probe/overlay/weave.go:
```diff
@@ -162,43 +336,120 @@ func (w *Weave) Report() (report.Report, error) {
+	}
+	// Peer corresponding to current host
+	if peer.Name == w.statusCache.Router.Name {
+		latests[report.HostNodeID] = w.hostID
```
I wonder if almost everything inside this conditional should be moved to yet another function? It's purpose isn't super clear to me, so giving it a name might help. Also, it would make the flow of `getPeerNode` more obvious to those who don't know what it's for.

2opremio replied:
I considered it during the process, but you need to pass all the locals (latests and node). Then decided it was not worth it. Now that it's grown quite a bit seems to make sense. Thanks

2opremio commented:
> I find the "#" notation a bit odd. I don't think we are using that elsewhere.

We are (in the container count of the container image topology). I tried to be consistent here but I can happily change it.

rade commented:
> We are (using the "#" notation) in the container count of the container image topology

The notation there is "# CONTAINERS: 2", not "CONTAINERS: #2".

2opremio commented:
> The notation there is "# CONTAINERS: 2", not "CONTAINERS: #2".

Ups, my bad, will fix

2opremio commented:
Everything should be addressed now addressed (and fixed a bug on the way). @jml Thanks for the review, plese take another look.

Here's a screenshot after the fixes:

<img width="2107" alt="screen shot 2016-11-03 at 23 02 02" src="https://cloud.githubusercontent.com/assets/2362916/19987148/e944a790-a219-11e6-91db-19bbb9e7f12c.png">

rade commented:
"DEF. SUBNET"? Why not spell it out? I recommend using *exactly* the same keys as in the `weave status` output, except perhaps for replacing "CompoundNoun" with "Compound Noun".

And why do the keys take up so much horizontal space, instead of leaving that to the values? Hence the unfortunate truncation of the proxy address.

2opremio commented on common/weave/client.go:
```diff
@@ -27,29 +27,47 @@ type Client interface {
+	}
+	Targets        []string
+	TrustedSubnets []string
+}
```
and since we couldn't vendor the whole data structure (that would imply vendoring `weave/prog/weaver/http.go` which is part of a program not a library) I decided to continue replicate the parts I needed.

2opremio replied:
I will be more than happy to refactor this once we can vendor the full data structure.
